src/custom/state/enhancedTransactions/hooks/index.ts

```diff
@@ -16,16 +16,14 @@ export type TransactionAdder = (params: AddTransactionHookParams) => void
  */
 export function useTransactionAdder(): TransactionAdder {
   const { chainId, account } = useWalletInfo()
-  const { gnosisSafeInfo } = useWalletDetails()
   const dispatch = useAppDispatch()
-
-  const isGnosisSafeWallet = !!gnosisSafeInfo
+  const isGnosisSafeApp = useIsGnosisSafeApp()
 
   return useCallback(
     (addTransactionParams: AddTransactionHookParams) => {
       if (!account || !chainId) return
 
-      const hashType = isGnosisSafeWallet ? HashType.GNOSIS_SAFE_TX : HashType.ETHEREUM_TX
```

Comment:
This should also work, right?

Are we sure we aren't covering an issue that makes us to not load the `gnosisSafeInfo` in some situations? shouldn't we have this object populated for the safeApp?

Reply:
Also, I'm just concerned this might break when you connect to the gnosis safe not through the app, since the hook `useIsGnosisSafeApp` is for Gnosis Safe App only and not native app using WC

Reply:
Good point, thanks!
I'm not sure and created another hook for this case: `useIsGnosisApp`